functions/src/modules/Users.functions.ts

```diff
@@ -0,0 +1,55 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+
+const moment = require('moment');
+const db = admin.database();
+
+/**
+ * Process registration of webform data
+ * 1. Registers user by transferring data into /users/ path
+ * 2. Check if users already exists
+ * 
+ * @function processRegistration()
+ */
+export const processRegistration = functions.database
+.ref('/webforms/registration/{registration_id}')
+.onCreate(async (snapshot, context) => {
+  const webform = snapshot.val();
+
+  const olduser = await db.ref("/users/").orderByChild("/emailAddress")
+                  .equalTo(webform.email_address).once("value");
+
+  // If the user already exists stop the execution and log the error
+  if (olduser.val() !== undefined) {
+    console.warn(`Warning! User already exist with email: ${webform.email_address}`)
+    return false;
+  }
+
+  // if user email doesn't exists prepare the new_user Object
+  const new_user: any = {
+    timestamp: moment.format('x'),
+    emailAddress: webform.email_address,
+    phoneNumber: webform.contact_number,
+    isAvailableOnWhatsApp: webform.i_am_available_on_whatsapp === 1,
+    location: webform.country,
+    name: webform.your_name,
+    services: (<any>Object).values(webform.seva).join(", "),
+    experience: webform.experience,
+    influencer: webform.where_did_u_hear_about_this_seva,
+    recommendedBy: webform.recommended_by,
+  }
+
+  /**
+   * Assigning user roles to new_user 
+   * Get the first letters of service name and assign it to @var new_user's roles
+   */
+  webform.seva.map(name => name.split(' ').map(piece => piece[0]).join(''))
+              .map(role => new_user.roles[role] = true);
+  
+  // assigning user languages to new_user Object 
+  webform.languages.map(lang => new_user.languages[lang] = true);
```

Comment:
I believe you should use [`reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) instead of `map` here. Map is not intended to have any side effect, and it is supposed to return a mapped array, which we don't need. Whereas you can use `reduce` to accumulate the roles in the object, and in this case you will be able to assign it directly to `newUser`.